app/src/main/java/fr/neamar/kiss/loader/LoadDocsPojos.java

```diff
@@ -0,0 +1,112 @@
+package fr.neamar.kiss.loader;
+
+import android.content.Context;
+import android.database.Cursor;
+import android.provider.MediaStore;
+
+import java.util.ArrayList;
+
+import fr.neamar.kiss.R;
+import fr.neamar.kiss.pojo.DocsPojo;
+
+import static android.provider.MediaStore.MediaColumns.DATA;
+
+public class LoadDocsPojos extends LoadPojos<DocsPojo> {
+
+    public LoadDocsPojos(Context context) {
+        super(context, "file://");
+    }
+
+    @Override
+    protected ArrayList<DocsPojo> doInBackground(Void... params) {
+        ArrayList<DocsPojo> docs = new ArrayList<>();
+
+        final String[] DOC_PROJECTION = {
+                MediaStore.Images.Media._ID,
+                MediaStore.Images.Media.DATA,
+                MediaStore.Files.FileColumns.MIME_TYPE,
+                MediaStore.Files.FileColumns.SIZE,
+                MediaStore.Images.Media.DATE_ADDED,
+                MediaStore.Files.FileColumns.TITLE
+        };
+
+        String[] selectionArgs = new String[]{".pdf", ".ppt", ".pptx", ".xlsx", ".xls", ".doc", ".docx", ".txt"};
```

Comment:
I wrote a dummyApp for measure performances of reading from .txt to fill hashMap and from .json to fill hashMap. Reading and parsing from .txt was better than jsonArray. So I decided to use reading from .txt file. And I did the examining a document' s step as you say which It makes sense to me too.
Thank you for your patience and interest.